test/optimizer/stats_calculator_test.cpp

```diff
@@ -0,0 +1,639 @@
+
+    NOISEPAGE_ASSERT(accessor_->GetSchema(table_oid_1_).GetColumns().size() == 1, "Table should have 1 column");
+    table_1_col_oid_ = accessor_->GetSchema(table_oid_1_).GetColumns().at(0).Oid();
+
+    NOISEPAGE_ASSERT(accessor_->GetSchema(table_oid_2_).GetColumns().size() == 2, "Table should have 2 column");
+    table_2_col_1_oid_ = accessor_->GetSchema(table_oid_2_).GetColumns().at(0).Oid();
+    table_2_col_2_oid_ = accessor_->GetSchema(table_oid_2_).GetColumns().at(1).Oid();
+  }
+};
+
+// NOLINTNEXTLINE
+TEST_F(StatsCalculatorTests, TestLogicalGet) {
+  RunQuery("INSERT INTO " + table_name_1_ + " VALUES(1), (NULL), (3);");
+  RunQuery("ANALYZE " + table_name_1_ + ";");
+  txn_manager_->Commit(test_txn_, transaction::TransactionUtil::EmptyCallback, nullptr);
+  test_txn_ = txn_manager_->BeginTransaction();
+
+  // Constructing logical get with no predicates
+  Operator logical_get =
+      LogicalGet::Make(test_db_oid_, table_oid_1_, {}, table_name_1_, false).RegisterWithTxnContext(test_txn_);
+  GroupExpression *gexpr = new GroupExpression(logical_get, {}, test_txn_);
+  gexpr->SetGroupID(group_id_t(1));
+  context_.GetMemo().InsertExpression(gexpr, false);
+
+  // CVE for column 1
+  parser::ColumnValueExpression col_a(table_name_1_, table_1_col_1_name_, test_db_oid_, table_oid_1_, table_1_col_oid_,
+                                      type::TypeId::INTEGER);
+  ExprSet required_cols;
+  required_cols.emplace(&col_a);
+
+  stats_calculator_.CalculateStats(gexpr, &context_);
+
+  auto *root_group = context_.GetMemo().GetGroupByID(gexpr->GetGroupID());
+  EXPECT_EQ(root_group->GetNumRows(), 3);
+  EXPECT_TRUE(root_group->HasNumRows());
+}
+
+// NOLINTNEXTLINE
+TEST_F(StatsCalculatorTests, TestInvalidLogicalGet) {
+  // Constructing logical get with no predicates
+  Operator logical_get =
+      LogicalGet::Make(test_db_oid_, catalog::INVALID_TABLE_OID, {}, "", false).RegisterWithTxnContext(test_txn_);
+  GroupExpression *gexpr = new GroupExpression(logical_get, {}, test_txn_);
+  gexpr->SetGroupID(group_id_t(1));
+  context_.GetMemo().InsertExpression(gexpr, false);
+
+  // CVE for column 1
+  parser::ColumnValueExpression col_a(table_name_1_, table_1_col_1_name_, test_db_oid_, table_oid_1_, table_1_col_oid_,
+                                      type::TypeId::INTEGER);
+  ExprSet required_cols;
+  required_cols.emplace(&col_a);
+
+  stats_calculator_.CalculateStats(gexpr, &context_);
+
+  auto *root_group = context_.GetMemo().GetGroupByID(gexpr->GetGroupID());
+  EXPECT_EQ(root_group->GetNumRows(), -1);
+  EXPECT_FALSE(root_group->HasNumRows());
+}
+
+// NOLINTNEXTLINE
+TEST_F(StatsCalculatorTests, TestNotPredicate) {
+  RunQuery("INSERT INTO " + table_name_1_ + " VALUES(1), (NULL), (3);");
+  RunQuery("ANALYZE " + table_name_1_ + ";");
+  txn_manager_->Commit(test_txn_, transaction::TransactionUtil::EmptyCallback, nullptr);
+  test_txn_ = txn_manager_->BeginTransaction();
+
+  // Constructing Logical Get with NOT EQUALS predicate
```

Comment:
Can you make the comment to be specific? For example, in this case it can be:
`Constructing Logical Get with NOT EQUALS predicate "NOT colA=1" from "empty_nullable_table"`
The same applies to other tests in this file.

Reply:
Should be fixed now